### DISCUSSION TITLE
Determine instance settings

### DISCUSSION THREAD

**Discussion by pcfriek1987** [apache/cloudstack#9626] [Q&A] [answered]
This was something that I just noticed today, how does it determine what settings it should show when looking at the setting of an instance?

For example, this one I got from an instance with Centos 7 and one with Debian 12.

<img width="1250" alt="Screenshot 2024-09-03 at 01 02 05" src="https://github.com/user-attachments/assets/90115b53-cec5-4765-808b-5610ca439c17">

This is one from a different instance but also with the Debian 12 template, very different options.

<img width="1264" alt="Screenshot 2024-09-03 at 01 04 02" src="https://github.com/user-attachments/assets/06af3b04-8179-4cbe-841f-7d9024db5be3">

**Comment by DaanHoogland** [comment]
@pcfriek1987 , these are settings set on the VM, some of them have defaults and some don't, so might be set by the user or by copying from the template on deploy.

**Reply by pcfriek1987** [reply]
So for example the nic adapter is not always changeable?

**Reply by BryanMLima** [reply, marked as answer]
Hey, @pcfriek1987, if it is a normal user, the global configuration `user.vm.denied.details` will filter the VM settings specified in this config. If the user has access to the API `updateVirtualMachine`, then it can add, remove and update any VM setting; though the VM cannot be running.
